BCP195bis/draft-ietf-uta-rfc7525bis.md

```diff
@@ -782,7 +782,7 @@ limits are maintained for each key:
 The latter applies to DTLS and QUIC but not to TLS itself, since TLS connections are torn down on the
 first decryption failure.
 
-When a sender is approaching CL, the implementation SHOULD initiate a new handshake (or in TLS 1.3, a Key Update) to rotate the session key.  When a receiver has reached IL, the implementation SHOULD close the connection.  Although these recommendations are a best practice, implementers need to be aware that it's not always easy to accomplish them in protocols that are built on top of TLS/DTLS or that re-use the TLS handshake protocol. For instance, in some protocols that depend on long-lived sessions it might not be possible to initiate a new TLS handshake or complete a TLS 1.3 Key Update without tearing down the old session and setting up a new one; this operation can introduce complexities such as draining all messages off the old session while setting up the new one or even maintaining two parallel sessions at the same time. As another example, the QUIC protocol (which, as noted, uses the TLS 1.3 handshake protocol with its own record layer) defines a method for Key Update as specified in {{Section 6 of RFC9001}}, but application protocols might not be able to emulate that method given their more constrained interation with TLS/DTLS.  As a result, these recommendations are not mandatory.
+When a sender is approaching CL, the implementation SHOULD initiate a new handshake (or in TLS 1.3, a KeyUpdatehandshake message) to rotate the session key.  When a receiver has reached IL, the implementation SHOULD close the connection.  Although these recommendations are a best practice, implementers need to be aware that it's not always easy to accomplish them in protocols that are built on top of TLS/DTLS or that re-use the TLS handshake protocol. For instance, in some protocols that depend on long-lived sessions it might not be possible to initiate a new TLS handshake or complete a TLS 1.3 KeyUpdate without tearing down the old session and setting up a new one; this operation can introduce complexities such as draining all messages off the old session while setting up the new one or even maintaining two parallel sessions at the same time. As another example, the QUIC protocol (which, as noted, uses the TLS 1.3 handshake protocol with its own record layer) defines a method for Key Update as specified in {{Section 6 of RFC9001}}, but application protocols might not be able to emulate that method given their more constrained interation with TLS/DTLS.  As a result, these recommendations are not mandatory.
```

Comment:
IIRC, Magnus suggested that we not reference draft-ietf-tsvwg-dtls-over-sctp-bis because that document could be encumbered with IPR. @thomas-fossati do you have specific concerns or suggestions on the text? IESG members looked at this and thought it was fine.

Reply:
> IIRC, Magnus suggested that we not reference draft-ietf-tsvwg-dtls-over-sctp-bis because that document could be encumbered with IPR.

I agree, there are sections there that can be referenced to describe the problem, without going into the details of the IPR'd solution.  But to be sure, let's stay clear of any possible trouble.

> @thomas-fossati do you have specific concerns or suggestions on the text? IESG members looked at this and thought it was fine.

I suggest to rephrase slightly your para:

```
When a sender is approaching CL, the implementation SHOULD initiate a new
handshake (in TLS 1.3, this can be achieved by sending a KeyUpdate message on
the established session) to rotate the session key. When a receiver has reached
IL, the implementation SHOULD close the connection. Although these
recommendations are a best practice, implementers need to be aware that it is
not always easy to accomplish them in protocols that are built on top of
TLS/DTLS without introducing coordination across layer boundaries.  See
{{Section 6 of RFC9001}} for an example of the cooperation needed in QUIC
between the crypto and transport layers to support key updates.  Note that in
general, application protocols might not be able to emulate that method given
their more constrained interaction with TLS/DTLS. As a result of these
complexities, these recommendations are not mandatory.
```

Reply:
Done!